From 442be39ada6db967117a381759c4346c08b86f9a Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 18 Dec 2019 14:25:46 +0200
Subject: [PATCH] Revert "[SW-0] gaudi: pci_pll: skip WA for metal fix asic fixup"

Since we remove the PLL W/A at all (no more old SoCs - only MF), we no
longer need to take special care about the W/A and PCIe PLL init.
Remove the code.

This reverts commit f69958bc99749b9754c42d0b21bf3488c8434f5d.

Change-Id: Ica817d931d9de7f72680dbde8e8091e90558ab2f
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/asic.c b/gaudi/asic.c
index 9f989e7..3936329 100644
--- a/gaudi/asic.c
+++ b/gaudi/asic.c
@@ -10,9 +10,6 @@
 #include <psoc_global_conf_regs.h>
 #include <psoc_pci_pll_regs.h>
 
-#include <hl_boot_if.h>
-#include <gaudi/gaudi_reg_map.h>
-
 #include <preboot.h>
 #include <pll.h>
 #include <uart.h>
@@ -272,18 +269,9 @@
 	return regmap_readl(mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_2) == 1;
 }
 
-static bool gaudi_pcie_pll_need_wa(void)
+static bool gaudi_pcie_strap_is_unset(void)
 {
 	u32 boot_strap = regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS);
-	u32 reset_cause = regmap_readl(mmPREBOOT_PCIE_EN); /* hard reset */
-
-	if (gaudi_reset_soc_done())
-		return false;
-
-	if (reset_cause == LKD_HARD_RESET_MAGIC) {
-		regmap_writel(0, mmPREBOOT_PCIE_EN);
-		return false;
-	}
 
 	/* Check if already done by BTL */
 	return !(boot_strap & 0x2);
@@ -293,7 +281,7 @@
 {
 	switch (cfg->base) {
 	case mmPSOC_PCI_PLL_BASE:
-		return gaudi_pcie_pll_need_wa();
+		return gaudi_pcie_strap_is_unset();
 	case mmPSOC_CPU_PLL_BASE:
 		return false;
 	case mmNIC2_HBM_PLL_BASE:
@@ -311,7 +299,7 @@
 {
 	int i, j;
 
-	if (!gaudi_pcie_pll_need_wa())
+	if (!gaudi_pcie_strap_is_unset())
 		return;
 
 	for (i = 0; i < ARRAY_SIZE(gaudi_pll_cfg); i++) {
diff --git a/gaudi/common.c b/gaudi/common.c
index abcbcb4..2d5174d 100644
--- a/gaudi/common.c
+++ b/gaudi/common.c
@@ -109,6 +109,7 @@
 
 	/* During hard reset triggered by LKD keep PCIe intact */
 	reset_cause = regmap_readl(mmPREBOOT_PCIE_EN);
+	regmap_writel(0, mmPREBOOT_PCIE_EN);
 	if (reset_cause == LKD_HARD_RESET_MAGIC)
 		return;
 
